airflow/providers/grafana/log/loki_task_handler.py

```diff
@@ -0,0 +1,242 @@
+    :param filename_template: template filename string (incase Loki is down)
+    :type filename_template: str
+    :param loki_conn_id: Connection ID that will be used for authorization to the Loki Platform.
+    :type loki_conn_id: str
+    :param name: the name of the custom log in Loki Logging. Defaults to 'airflow'.
+    :type name: str
+    :param labels: (Optional) Mapping of labels for the entry.
+    :type labels: dict
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        base_log_folder: str,
+        filename_template: str,
+        loki_conn_id: str,
+        name: str = DEFAULT_LOGGER_NAME,
+        labels: Optional[Dict[str, str]] = None,
+    ):
+        super().__init__(base_log_folder, filename_template)
+        self.loki_conn_id = loki_conn_id
+        self.name: str = name
+        self.timestamp_pattern = "%Y-%m-%dT%H:%M:%S"
+        self.labels = labels
+        self._session: Optional[requests.Session] = None
+
+    @cached_property
+    def get_conn(self):
+        """Loki connection for client"""
+        return BaseHook.get_connection(self.loki_conn_id)
+
+    @property
+    def session(self) -> requests.Session:
+        """Create HTTP session"""
+        if self._session is None:
+            self._session = requests.Session()
+            self._session.auth = (self.get_conn.login, self.get_conn.password) or None
+        return self._session
+
+    def is_loki_alive(self):
+        """Checks whether Loki is ready for pushing/pulling logs"""
+        try:
+            status = self.session.get(
+                f"{self.get_conn.host}/ready",
+            )
+            return status.status_code
+        except ConnectionError as error_msg:
+            self.log.exception(error_msg)
+            return None
+
+    @staticmethod
+    def _task_label(task_instance: TaskInstance) -> Dict[str, str]:
+        """
+        Returns task instance labels for Loki which will use while reading
+        and writing logs from loki.
+        :param task_instance: task instance object
+        :type: task_instance: TaskInstance
+        """
+        # Not adding execution date since it violates Loki label standards
+        # https://grafana.com/blog/2020/08/27/the-concise-guide-to-labels-in-loki/
+
+        return {
+            "airflow_dag_id": task_instance.dag_id,
+            "airflow_task_id": task_instance.task_id,
+            "airflow_try_number": str(task_instance.try_number),
+        }
+
+    def get_label(self, task_instance: TaskInstance) -> Dict[str, str]:
+        """
+        Update task_labels with optional labels and return Loki labels.
+        :param task_instance: task instance object
+        :type: task_instance: TaskInstance
+        """
+        tags = {}
+        task_labels = self._task_label(task_instance)
+        if self.labels:
+            tags.update(self.labels)
+            tags.update(task_labels)
+            return tags
+        return task_labels
```

Comment:
```suggestion
        if self.labels:
            tags = copy.copy(self.labels)
        else:
            tags = {}
        tags.update(self._task_label(task_instance))
        return tags
```

You'll need to `import copy` too.